services/app-api/handlers/rate/calculations/adminstrativeCalculation.ts

```diff
@@ -0,0 +1,74 @@
+import { Measure, Program } from "../../../types";
+import { RateCalculation } from "./rateCalculation";
+import { DataSource, FormattedMeasureData, UniqMeasureAbbr } from "./types";
+
+export class AdminstrativeCalculation extends RateCalculation {
+  dataSrcMap = [
+    {
+      Medicaid: [DataSource.Administrative],
+      CHIP: [DataSource.Administrative],
+    },
+    {
+      Medicaid: [DataSource.Administrative],
+      CHIP: [DataSource.EHR],
+    },
+    {
+      Medicaid: [DataSource.EHR],
+      CHIP: [DataSource.EHR],
+    },
+    {
+      Medicaid: [DataSource.EHR],
+      CHIP: [DataSource.Administrative],
+    },
+    {
+      Medicaid: [DataSource.Administrative, DataSource.EHR],
+      CHIP: [DataSource.Administrative],
+    },
+  ];
+  check(arr: FormattedMeasureData[]): boolean {
+    const chipSources =
+      arr.find((data) => data.column === "CHIP")?.dataSource ?? [];
+    const medicaidSources =
+      arr.find((data) => data.column === "Medicaid")?.dataSource ?? [];
+
+    //if the user had selected hybrid as a data source, we will not use this calculation
+    const isHybrid = [chipSources, medicaidSources].some((srcs) => {
+      return (srcs as string[])?.includes(DataSource.Hybrid);
+    });
+
+    if (!isHybrid) {
+      for (var i = 0; i < this.dataSrcMap.length; i++) {
+        const dataSrc = this.dataSrcMap[i];
```

Comment:
`for..of` is generally preferable to both `for`, `for..in`, and `.forEach`, when possible. Which, here, it is 🤠 

```suggestion
      for (const dataSrc of this.dataSrcMap) {
```